src/stores/DataStore.js

```diff
@@ -12,7 +12,14 @@ class DataStore extends Collection {
   }
 
   // Stubs
-  create() { return undefined; }
+  create(data, cache = true) {
+    const existing = this.get(data.id);
+    if (existing) return existing;
+
+    const entry = new this.holds(this.client, data);
```

Comment:
Wouldn't this cause an issue when you pass an iterable to the store?
`this.holds` would be undefined until the deriving class assigns it which woud happen after creating from the iterable.

Reply:
easy fix is to change the constructor to take the `holds` param

Reply:
also perhaps if i just make a DataStore without holds it won't try to instantiate anything and just store the object directly, that seems sensible.